### DISCUSSION TITLE
what to do with saved IR

### DISCUSSION THREAD

**Discussion by regehr** [AliveToolkit/alive2#825] [Ideas]
ok now that that "save IR" patch has landed, I wanted to sync up and see if we agree on what to do next. I have two things on my list:

1. a script that looks through a log directory (left over from building a project using alivecc) and auto-reduces all .bc files found there. this should hopefully be easy now.
2. the whole unattended fuzzing setup using Yuyou's tool. this requires a good set of seeds for mutation, which I've not worked on yet, and then some automated reduction and deduplication.

I think these two things are probably separate, but maybe not...

**Comment by nunoplopes** [comment]
A missing bit is how to map pass names back to opt cmd arguments. If one uses -O2, we need to map "Instruction Combiner" back to -instcombine.
The info is all in the LLVM sources, we can parse it, e.g.:
```c++
INITIALIZE_PASS_BEGIN(NewGVNLegacyPass, "newgvn", "Global Value Numbering",
                      false, false)
```

The hard bit, which I don't know how to recover, is the analyses uses. For example, the repro cmd line may need `-basicaa` or something like. Let's call this v2. Not super important for a 1st version.

**Comment by regehr** [comment]
yep
```
my %PASSMAP = (
    "LoopVectorizePass" => "-loop-vectorize",
    "SROAPass" => "-sroa",
    "SimplifyCFGPass" => "-simplifycfg",
    );
```
